src/layers/core/line-layer/line-layer-vertex.glsl

```diff
@@ -26,21 +26,30 @@ attribute vec3 instanceTargetPositions;
 attribute vec4 instanceColors;
 attribute vec3 instancePickingColors;
 
+uniform vec2 screenSize;
+uniform float strokeWidth;
 uniform float opacity;
 uniform float renderPickingBuffer;
 
 varying vec4 vColor;
 
 void main(void) {
   // Position
-  vec3 source = project_position(instanceSourcePositions);
-  vec3 target = project_position(instanceTargetPositions);
+  vec3 sourcePos = project_position(instanceSourcePositions);
+  vec3 targetPos = project_position(instanceTargetPositions);
+  vec4 source = project_to_clipspace(vec4(sourcePos, 1.0));
+  vec4 target = project_to_clipspace(vec4(targetPos, 1.0));
 
   // linear interpolation of source & target to pick right coord
   float segmentIndex = positions.x;
-  vec3 p = mix(source, target, segmentIndex);
+  vec4 p = mix(source, target, segmentIndex);
 
-  gl_Position = project_to_clipspace(vec4(p, 1.));
+  // extrude by strokeWidth
+  vec2 dir = normalize((target.xy - source.xy) * screenSize);
+  vec2 perp = vec2(-dir.y, dir.x);
+  vec2 offset = perp * positions.y * strokeWidth / screenSize;
```

Comment:
My first thought was that this always extrudes to one size (rather than centered?) If so it would probably only look good for smaller `strokeWidths`. If this is not true please add some comment explaining how the centering works.